# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.cpp
# Commit: 897f6873d4c1
# Full Hash: 897f6873d4c12698dab9b8d7601e167604b2a3cf
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-06-22 21:47:58
# Description:
#   Bug 1646328 - Downgrade sameProcessAsFrameLoader/remoteType to a debug assert, r=mattwoodrow
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D80259
# ==============================================================================

diff -r 9a8b5b9ca323 -r 897f6873d4c1 dom/base/nsFrameLoader.cpp
--- a/dom/base/nsFrameLoader.cpp	Fri Jun 19 19:23:15 2020 +0000
+++ b/dom/base/nsFrameLoader.cpp	Mon Jun 22 11:56:29 2020 +0000
@@ -399,9 +399,13 @@
     return;
   }
   RefPtr<ContentParent> contentParent = browserParent->Manager();
-  MOZ_DIAGNOSTIC_ASSERT(
-      !hasRemoteType || contentParent->GetRemoteType() == aRemoteType,
-      "If specified, remoteType attribute must match sameProcessAsFrameLoader");
+
+  // NOTE: This assertion is known to fail in the wild. It is being kept around
+  // only to ensure that we don't land and test new code which breaks this
+  // desired invariant. (bug 1646328)
+  MOZ_ASSERT(!hasRemoteType || contentParent->GetRemoteType() == aRemoteType,
+             "If specified, remoteType attribute should match "
+             "sameProcessAsFrameLoader");
   aRemoteType = contentParent->GetRemoteType();
   *aChildID = contentParent->ChildID();
 }
